### ISSUE TITLE
can you add some functions to BAM.Reader?

### ISSUE THREAD

**Issue by panxiaoguang** [BioJulia/XAM.jl#39]
can you add some new functions to the whole BAM class? for examples,

- get mapped and unmaped read numbers for every reference 
- get all reference names
- ...
- check if one reference name  was in bam files

**Comment by CiaranOMara** [comment]
The functions you are requesting already exist. You can compose with functions from Julia's [Base and Standard Library](https://docs.julialang.org/en/v1/).

```julia
filepath = "data.bam"
reader = open(BAM.Reader, filepath, index=filepath * ".bai")

# get mapped and unmaped read numbers for every reference
mapped = Iterators.filter(BAM.ismapped, reader) |> collect
unmapped = Iterators.filter(!BAM.ismapped, reader) |> collect

# get all reference names
reader.refseqnames

# check if one reference name was in bam files
in("chr1", reader.refseqnames)
```

**Comment by panxiaoguang** [comment]
Thank you very much! But I still have some questions?
I readed the document of XAM on juliahub, I did't see any method about reader on API reference sectors, only have SAM API and BAM API.

another question is for calculating the mapped numbers, I noticed you you still use the method for BAM_record to calculate the read number ,maybe it will iterater all record(reads). I think it will use many times to do this work. In fact, the number of reads per reference has been calculated in advance and stored in the index file. 
So, maybe we can use 

```julia
idx = BAM.BAI("data.bam.bai")
map(x->x[3].n_mapped,idx.index.data)
```

But in this way, we can't know the corresponding reference, we can only use the order of reference_id to guess. If some refs have no read, an error will be reported.
